### PR TITLE
Fixes nightly issues and caching in CI

### PR DESCRIPTION
**Pull request by kelsos**
- Introduces output folding for build scripts on github actions #2153
- Add a check in the hook to fix the infinite loop @LefterisJP if you have any comments? I tested the Linux binary it worked, I also run the nightlies they work fine. https://github.com/kelsos/rotki/runs/1748644073?check_suite_focus=true
- Adds a new tag detection on the `check` action. Now if the commit contains [run all] it will skip the automatic detection and will proceed to run all the jobs. This should be useful for example when we are doing some configuration change or something.

### CODE REVIEW

File: tools/pyinstaller_hooks/hook-rotkehlchen.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import os
-
 import pkg_resources
 
 datas = []
+processed = []
```
Comment (LefterisJP) [inline tools/pyinstaller_hooks/hook-rotkehlchen.py:5]: as discussed in discord can you make this into a python set?

Also can you add a comment here explaining why this is needed (circular package dependencies)

**Comment by kelsos** [comment]
@LefterisJP the change is done

**Comment by LefterisJP** [comment]
Look at you! Becoming a python wizard
tenor (1)

**Review by LefterisJP — APPROVED** [review]
LGTM. Thanks a lot for handling this Kelsos.

**Pull request merged**